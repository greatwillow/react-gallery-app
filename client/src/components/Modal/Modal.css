.screen-overlay {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  padding-top: 50px;
  background-color: rgba(0, 0, 0, 0.7);
  z-index: 6;
  overflow: scroll;
}

.modal {
  position: absolute;
  left: 50%;
  -webkit-transform: translate(-50%, 0);
  -ms-transform: translate(-50%, 0);
  transform: translate(-50%, 0);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  width: 70%;
  height: auto;
  background-color: #fff;
}

.modal-close-button {
  position: absolute;
  top: 15px;
  right: 15px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 20px;
  height: 20px;
  color: #000;
  border: none;
  background-color: transparent;
  font-size: 20px;
  cursor: pointer;
}

@media (max-width: 700px) {
  .modal {
    width: 90%;
  }
}

@media (max-width: 500px) {
  .modal {
    width: 100%;
  }
}
