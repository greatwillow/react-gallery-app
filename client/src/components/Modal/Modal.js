import React, { Component } from 'react';
import PropTypes from 'prop-types';
// STYLES
import './Modal.css';
import 'font-awesome/css/font-awesome.css';

class Modal extends Component {
  componentDidMount() {
    document.addEventListener('mousedown', this.handleClickOutside);
  }

  componentWillUnmount() {
    document.removeEventListener('mousedown', this.handleClickOutside);
  }

  handleClickOutside = event => {
    if (this.modalRef && !this.modalRef.contains(event.target)) {
      this.props.handleShowModal();
    }
  };

  render() {
    console.log('MODAL PHOTO ', this.props.photo);
    return (
      <div className="screen-overlay">
        <div
          ref={element => {
            this.modalRef = element;
          }}
          className="modal"
        >
          <button className="modal-close-button" onClick={this.props.handleShowModal}>
            <i className="fa fa-close" />
          </button>
          <img src={this.props.photo.src.large} alt={this.props.photo.url} />

          <h3>Photo taken by: </h3>
          <p>{this.props.photo.photographer}</p>
        </div>
      </div>
    );
  }
}

Modal.propTypes = {
  handleShowModal: PropTypes.func.isRequired,
  photo: PropTypes.object.isRequired
};

export default Modal;
