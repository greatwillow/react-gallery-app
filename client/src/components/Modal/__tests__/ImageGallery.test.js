import React from 'react';
import Modal from '../Modal';
import renderer from 'react-test-renderer';

describe('Modal component renders the modal correctly', () => {
  const handleShowModal = jest.fn();
  const handleSearchTermChange = jest.fn();
  const photo = {
    src: {
      medium: null
    }
  };
  const windowWidth = 800;

  const rendered = renderer.create(
    <Modal handleShowModal={handleShowModal} photo={photo} windowWidth={windowWidth} />
  );
  it('renders correctly', () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
