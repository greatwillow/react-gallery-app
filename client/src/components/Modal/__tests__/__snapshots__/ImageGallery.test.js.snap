// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders the modal correctly renders correctly 1`] = `
<div
  className="screen-overlay"
>
  <div
    className="modal"
  >
    <button
      className="modal-close-button"
      onClick={[Function]}
    >
      <i
        className="fa fa-close"
      />
    </button>
    <img
      alt={undefined}
      src={null}
      width="50%"
    />
    <p>
      This photograph has been taken from
       
      <a
        href="http://www.pexels.com"
      >
        www.pexels.com
      </a>
      .
      <br />
      The stated photographer is: Â 
      <br />
      Find out more by clicking 
      <a
        href={undefined}
      >
        here.
      </a>
    </p>
  </div>
</div>
`;
